File: CourseCRUD-SPA/src/app/_core/utilities/search-param.ts
export interface SearchParam {
  sortParams: SortParam[];
  filterParam: FilterParam;
}

export interface SortParam {
  sortColumn: string;
  sortBy: SortBy;
  sortClass: SortClass;
}

export interface FilterParam {
  keyword: string;
  category_ID: number | null;
  price: number | null;
}

export enum SortBy {
  Asc = 'ASC',
  Desc = 'DESC'
};

export enum SortClass {
  Asc = 'fa fa-sort-amount-asc ml-3',
  Desc = 'fa fa-sort-amount-desc ml-3'
};

File: CourseCRUD-API/Helpers/Utilities/SearchParam.cs
using System.Collections.Generic;

namespace CourseCRUD_API.Helpers.Utilities
{
    public class SearchParam
    {
        public List<SortParam> SortParams { get; set; }
        public FilterParam FilterParam { get; set; }
    }

    public class SortParam
    {
        public string SortColumn { get; set; }
        public string SortBy { get; set; }
    }

    public class FilterParam
    {
        public string Keyword { get; set; }
        public int? Category_ID { get; set; }
        public decimal? Price { get; set; }
    }
}